docs/source/clamav/clamav.rst

```diff
@@ -0,0 +1,287 @@
+--------------------------------------
+
+The recommended method to install ClamAV on an NI Linux Real-Time system is through NILRT-SNAC,
+which handles the complete installation and configuration automatically:
+
+.. code:: bash
+    
+    nilrt-snac configure
+
+This automated installation process:
+
+- Installs ClamAV packages from pre-built IPK files
+- Configures virus database directories with proper permissions
+- Sets up logging with appropriate ownership and permissions
+- Restores DNS services if broken during installation
+- Creates the ``clamav-scan`` wrapper script with automatic swap file creation for scanning large files
+
+.. note::
+    The automated installation uses ``nilrt-snac configure`` which takes care of DNS service restoration,
+    and adds patches to create swap files for big file scanning.
+    This ensures ClamAV works reliably on resource-constrained embedded systems.
+
+Manual Installation (Alternative)
+----------------------------------
+
+If you prefer to install ClamAV manually, you can use opkg directly:
+
+.. code:: bash
+
+    opkg install clamav clamav-freshclam
+
+However, manual installation requires additional configuration steps that are automatically handled by
+NILRT-SNAC. The automated method is strongly recommended.
+
+Verifying Installation
+----------------------
+
+After installation, verify that ClamAV is properly installed and configured:
+
+**Quick Version Check:**
+
+.. code:: bash
+
+    clamscan --version
+
+You should see output showing the ClamAV version (1.4.3 or newer) and configuration paths.
+
+**Comprehensive Verification:**
+
+Use NILRT-SNAC's verify command to perform a complete system check:
+
+.. code:: bash
+
+    nilrt-snac verify
+
+This command checks:
+
+- ClamAV package installation status
+- Virus signature database integrity (checks for corrupted signatures)
+- Missing or incomplete signature files
+- Configuration file validity
+- Directory permissions and ownership
+- Service status and functionality
+
+@tbd : The verify command will report any issues found and provide guidance on how to resolve them.
+
+Configuration
+=============
+
+ClamAV uses configuration files to control its behavior for virus scanning and signature updates.
+The main configuration files are:
+
+- ``/usr/etc/freshclam.conf``: Configuration for the freshclam signature updater
+- ``/etc/clamav/clamd.conf``: Configuration for the ClamAV daemon (optional)
+- ``/var/lib/clamav/``: Directory containing virus signature databases
+
+When installed via NILRT-SNAC, these files are automatically configured with appropriate settings.
+
+Updating Virus Signatures
+--------------------------
+
+Before scanning files, you need to download the latest virus signature databases. ClamAV provides the following methods to update signatures:
+
+**Method 1: Online Update (Recommended)**
+
+If your target has internet connectivity, update signatures using freshclam:
+
+.. code:: bash
+
+    sudo freshclam
```

Comment:
Let's remove the `sudo` references from this codeblock and elsewhere. Mainline NILRT users normally don't use sudo (they run as UID 0). And general best practice for linux documentation is not to include 'sudo' references for commands that need to be run as root, because you don't *really* know how the user's permissions are setup. Instead, just say in the description "run freshclam with superuser permissions".